### PR TITLE
Attribute-based variant picker

### LINKED ISSUE #614: Introduce attribute based variant picker
Variant picker should allow to select variants by attributes for example color and size.
Variant data should be exposed by an API endpoint or by view context.

image

For MVP purposes I suggest following approach:
- Create React component that renders radio group for each variant attribute
- Component should update price automatically 
- When product is unavailable or particular variant combination does not exist we should display information that product is unavailable.

### PR DESCRIPTION
**Pull request by elwoodxblues**
Fixes #614

### CODE REVIEW

File: saleor/product/models/utils.py [hunk 1]
```diff
@@ -1,12 +1,10 @@
 def get_attributes_display_map(variant, attributes):
     display = {}
     for attribute in attributes:
-        value = variant.get_attribute(attribute.pk)
+        value = variant.attributes.get(str(attribute.pk))
```
Comment (patrys) [inline saleor/product/models/utils.py]: `str` or `smart_text`?

**Comment by artursmet** [comment]
Looks good, for MVP purposes:

image

There's a radio group for each variant level attribute, product attributes are not displayed at all. I think we can display particular product attributes somewhere above variant selector.

In next version we can introduce variant selector that is aware of other versions of product (with different product-level attributes set)

**Comment by patrys** [comment]
I have a feeling this should be rebased on top of whatever lands to fix #644

**Comment by patrys** [comment]
CI seems to fail because `npm-shrinkwrap.json` was not regenerated properly. Did you use `yarn` to install the new package?

**Comment by elwoodxblues** [comment]
Yes I did. Will fix.

**Comment by patrys** [comment]
Could you try using a newer version of `npm`? It seems the entire shrinkwrap file has changed.

**Comment by patrys** [comment]
This introduces a lot of cosmetic changed like adding semicolons that we did not previously use. Can we separate these changes? My current `eslint` config highlights all the semicolons in red.

**Comment by elwoodxblues** [comment]
@patrys We should add default `eslint` config to the project, that would help with developing React code and JS generally. I can add it to this PR. Question whether to require semicolons or not is open. I would personally use them.

**Comment by patrys** [comment]
I agree that we should add an eslint config but what I'm asking of this PR is to exclude cosmetic changes. We can add semicolons in a separate change if we agree that we want them to stay.

**Comment by elwoodxblues** [comment]
@patrys Done

**Pull request merged**